src/operations/txbuild.js

```diff
@@ -478,6 +702,177 @@ function makeRenewal(fullyQualifiedName: string,
     })
 }
 
+
+/**
+ * Generates a namespace preorder transaction for a namespace
+ * @param {String} namespaceID - the namespace to pre-order
+ * @param {String} revealAddress - the address to receive the namespace (this
+ *    must be passed as the 'revealAddress' in the namespace-reveal transaction)
+ * @param {String} paymentKeyHex - a hex string of the private key used to
+ *    fund the transaction
+ * @returns {Promise} - a promise which resolves to the hex-encoded transaction.
+ *    this function *does not* perform the requisite safety checks -- please see
+ *    the safety module for those.
+ * @private
+ */
+function makeNamespacePreorder(namespaceID: string,
+                               revealAddress: string,
+                               paymentKeyHex: string) {
+  const network = config.network
+
+  const paymentKey = hexStringToECPair(paymentKeyHex)
+  const preorderAddress = paymentKey.getAddress()
+
+  const preorderPromise = Promise.all([network.getConsensusHash(),
+                                       network.getNamespacePrice(namespaceID)])
+        .then(([consensusHash, namespacePrice]) =>
+          makeNamespacePreorderSkeleton(
+            namespaceID, consensusHash,
+            network.coerceAddress(preorderAddress),
```

Comment:
We decided at some point (actually found it, search on discourse forums is amazing! https://forum.blockstack.org/t/2018-01-18-engineering-meeting/3979/1) that we didn't want to coerce network addresses, because _long-term_ we want clients to have to know the difference between when they're operating on mainnet versus regtest.